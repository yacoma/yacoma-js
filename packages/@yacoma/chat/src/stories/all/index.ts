export * from './Explorer'
// export * from './Header'
// export * from './Chat'
// export * from './People'
// export * from './ContactsWidget'
//
