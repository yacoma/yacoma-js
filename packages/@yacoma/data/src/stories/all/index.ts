export * from './ContactPrint'
export * from './Loading'
export * from './RegisterName'
export * from './RegisterVerify'
export * from './RegisterWords'
export * from './WordLogin'
