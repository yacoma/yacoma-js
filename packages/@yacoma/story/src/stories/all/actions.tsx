import { action, IOperator } from 'overmind'
import * as React from 'react'
import styled from 'styled-components'
import { Stories } from '../..'
import { Comp } from '../../app'
import { createHook } from 'overmind-react'

interface Config {
  state: {
    barName: string
  }
  actions: {
    doSomething: Operator<string>
  }
  effects: {}
}

type Operator<Input = any, Output = Input> = IOperator<Config, Input, Output>

const doSomething: Operator<string> = action(({ state }, value) => {
  state.barName = value
})

const config = {
  state: {
    barName: 'Mona',
  },
  actions: { doSomething },
}
const useOvermind = createHook<typeof config>()

const Box = styled.div`
  border: 1px solid #999;
  padding: 3px;
  cursor: pointer;
  background: orange;
`

const myComponent: Comp<{ className?: string }> = ({ className }) => {
  const app = useOvermind()
  return (
    <Box
      className={className}
      onClick={() => {
        app.actions.doSomething(app.state.barName === 'Lisa' ? 'Mona' : 'Lisa')
      }}
    >
      {app.state.barName}
    </Box>
  )
}

export const sequencesStories: Stories = {
  name: 'actions',
  stories: [
    {
      name: 'simple',
      component: myComponent,
      config,
    },
  ],
}
