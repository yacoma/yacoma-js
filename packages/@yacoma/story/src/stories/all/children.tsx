import { createHook } from 'overmind-react'
import * as React from 'react'
import { Stories } from '../..'
import { Comp, styled } from '../../app'

interface Config {
  state: {
    foo: string
  }
}
const useOvermind = createHook<Config>()

const Foo: Comp = () => {
  const app = useOvermind()
  return <div>{app.state.foo}</div>
}

const Orange = styled.div`
  padding: 10px;
  background: orange;
`

const component: React.SFC<{}> = function ShowChildren({ children }) {
  return <Orange>{children}</Orange>
}

export const children: Stories<{ name: string }> = {
  name: 'children',
  config: { state: { foo: 'fool' } },
  component,
  stories: [
    {
      name: 'simple',
      children: [<div key="1">Hello</div>, <div key="2">World</div>],
    },
    {
      name: 'with tracking',
      children: [
        <Foo key="1" />,
        <div key="2">Hello</div>,
        <div key="3">World</div>,
      ],
    },
  ],
}
