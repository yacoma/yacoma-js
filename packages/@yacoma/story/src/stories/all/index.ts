export * from './actions'
export * from './children'
export * from './props'
export * from './state'
export * from './storiesDefaults'
