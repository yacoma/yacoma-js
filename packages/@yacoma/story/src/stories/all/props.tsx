import { action } from 'overmind'
import { createHook } from 'overmind-react'
import { build } from '@yacoma/build'
import { TStories } from '../..'
import { Comp } from '../../app'

declare module 'overmind' {
  // eslint-disable-next-line @typescript-eslint/interface-name-prefix
  interface IState extends Config {}
}

const config = build({
  name: 'app',
  state: { foo: { bar: { baz: 'baz' } } },
}).config()

const useOvermind = createHook<typeof config>()

const component: Comp<{ name: string }> = ({ name }) => {
  const app = useOvermind()
  return (
    <div>
      This is {name} ({app.state.foo.bar.baz})
    </div>
  )
}

const anon = () => {
  return <div>simple div</div>
}

export const propsStories: TStories<
  typeof config,
  {
    name: string
    bool?: boolean
    callback?: () => void
    tag?: any
    number?: number
  }
> = {
  name: 'props',
  config,
  stories: [
    {
      name: 'simple',
      component,
      props: { name: 'foo' },
    },
    {
      name: 'state change on click',
      component,
      titleClick: action(({ state }) => {
        state.foo.bar.baz = state.foo.bar.baz === 'Sweet' ? 'Home' : 'Sweet'
      }),
      props: { name: 'bar' },
    },
    {
      name: 'props as function',
      component,
      props: ({ state }) => ({
        name: 'bar',
        bool: true,
        callback() {},
        tag: state.foo.bar,
        number: 14,
      }),
    },
    {
      name: 'plain component',
      component: function Foo() {
        return <div>simple div</div>
      },
    },
    {
      name: 'component without name',
      component: anon,
    },
  ],
}
