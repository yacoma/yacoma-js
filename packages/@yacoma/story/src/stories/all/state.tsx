import { createHook } from 'overmind-react'
import * as React from 'react'
import { themeProxy } from '@yacoma/theme'
import { Stories } from '../..'
import { Comp, styled } from '../../app'

interface Config {
  state: {
    things: {
      [key: string]: {
        name: string
      }
    }
    otherThings: {
      [key: string]: {
        name: string
      }
    }
    something: string
  }
}

const useOvermind = createHook<Config>()

const selectedTheme = {
  textColor: '#fd50b8',
}

const theming = themeProxy(selectedTheme)

const Wrapper = styled.div`
  color: ${theming.textColor};
  border: 1px solid ${theming.textColor};
  padding: 3px;
`

const component: Comp<{
  className?: string
  id: string
}> = ({ className, id }) => {
  const app = useOvermind()
  return <Wrapper className={className}>{app.state.things[id].name}</Wrapper>
}

export const stateStories: Stories = {
  name: 'state',
  config: {
    state: {
      // To test merging object in StoryDisplay
      otherThings: {},
      // To test theme is used
      theme: {
        selectedTheme,
      },
    },
  },
  stories: [
    {
      name: 'simple',
      component,
      state: {
        things: {
          difool: { name: 'I am John Difool' },
        },
        otherThings: {},
        something: 'hello',
      },
      props: { id: 'difool' },
    },
  ],
}
