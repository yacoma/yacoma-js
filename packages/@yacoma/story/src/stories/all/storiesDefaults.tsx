import { action } from 'overmind'
import * as React from 'react'
import { Stories } from '../..'
import { Comp } from '../../app'

const DefaultComponent: Comp<{ name: string }> = ({ name }) => (
  <div>This is {name}</div>
)

const wrapper: Comp = ({ children }) => <div>{children}</div>

export const storiesDefaults: Stories<{ name: string }> = {
  name: 'defaults in Stories',
  props: { name: 'foo' },
  component: DefaultComponent,
  wrapper,
  titleClick: action(() => alert('Title click')),
  stories: [
    {
      name: 'simple all defaults',
    },
  ],
}
