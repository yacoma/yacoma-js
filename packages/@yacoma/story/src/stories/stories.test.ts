import { testStories } from '..'
import * as stories from './all'

testStories({}, stories)
