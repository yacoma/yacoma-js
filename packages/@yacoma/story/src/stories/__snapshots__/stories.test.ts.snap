// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should render simple 1`] = `
<div
  className="theme-actions-simple"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          simple
        </div>
        <div
          className="sc-gqjmRU dHWrsZ"
        >
          &lt;My /&gt;
          <div
            className="jsx"
          >
            &lt;My /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH dNRbgB"
      >
        <div
          className="sc-dnqmqq fJIfVz"
          onClick={[Function]}
        >
          Mona
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`children should render simple 1`] = `
<div
  className="theme-children-simple"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          simple
        </div>
        <div
          className="sc-VigVT jgQSXV"
        >
          &lt;ShowChildren&gt;...
          <div
            className="jsx"
          >
            &lt;ShowChildren&gt;
  &lt;div&gt;Hello&lt;/div&gt;
  &lt;div&gt;World&lt;/div&gt;
&lt;/ShowChildren&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH dNRbgB"
      >
        <div
          className="sc-iwsKbI fPfSwy"
        >
          <div>
            Hello
          </div>
          <div>
            World
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`children should render with tracking 1`] = `
<div
  className="theme-children-with-tracking"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          with tracking
        </div>
        <div
          className="sc-jTzLTM jfpFEd"
        >
          &lt;ShowChildren&gt;...
          <div
            className="jsx"
          >
            &lt;ShowChildren&gt;
  &lt;Foo /&gt;
  &lt;div&gt;Hello&lt;/div&gt;
  &lt;div&gt;World&lt;/div&gt;
&lt;/ShowChildren&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH dNRbgB"
      >
        <div
          className="sc-iwsKbI fPfSwy"
        >
          <div>
            fool
          </div>
          <div>
            Hello
          </div>
          <div>
            World
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`defaults in Stories should render simple all defaults 1`] = `
<div
  className="theme-defaults-in-Stories-simple-all-defaults"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          simple all defaults
        </div>
        <div
          className="sc-kGXeez hOLhYv"
        >
          &lt;DefaultComponent name="foo" /...
          <div
            className="jsx"
          >
            &lt;DefaultComponent name="foo" /&gt;
          </div>
        </div>
      </div>
      <div>
        <div>
          This is 
          foo
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`props should render component without name 1`] = `
<div
  className="theme-props-component-without-name"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          component without name
        </div>
        <div
          className="sc-chPdSV fKVxFa"
        >
          &lt;anon /&gt;
          <div
            className="jsx"
          >
            &lt;anon /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH dNRbgB"
      >
        <div>
          simple div
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`props should render plain component 1`] = `
<div
  className="theme-props-plain-component"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          plain component
        </div>
        <div
          className="sc-kAzzGY gJpnWy"
        >
          &lt;Foo /&gt;
          <div
            className="jsx"
          >
            &lt;Foo /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH dNRbgB"
      >
        <div>
          simple div
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`props should render props as function 1`] = `
<div
  className="theme-props-props-as-function"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          props as function
        </div>
        <div
          className="sc-cSHVUG ccyOgf"
        >
          &lt;DoSomething name="bar" bool c...
          <div
            className="jsx"
          >
            &lt;DoSomething name="bar" bool callback={function callback} tag={app.state.foo.bar} number={14} /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH dNRbgB"
      >
        <div>
          This is 
          bar
           (
          baz
          )
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`props should render simple 1`] = `
<div
  className="theme-props-simple"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          simple
        </div>
        <div
          className="sc-fjdhpX jdGklJ"
        >
          &lt;DoSomething name="foo" /&gt;
          <div
            className="jsx"
          >
            &lt;DoSomething name="foo" /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH dNRbgB"
      >
        <div>
          This is 
          foo
           (
          baz
          )
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`props should render state change on click 1`] = `
<div
  className="theme-props-state-change-on-click"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          state change on click
        </div>
        <div
          className="sc-jzJRlG cgbAdd"
        >
          &lt;DoSomething name="bar" /&gt;
          <div
            className="jsx"
          >
            &lt;DoSomething name="bar" /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH dNRbgB"
      >
        <div>
          This is 
          bar
           (
          baz
          )
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`state should render simple 1`] = `
<div
  className="theme-state-simple"
>
  <div
    className="sc-ifAKCX hYWxfb"
  >
    <div
      className="sc-htpNat hIwNRt"
    >
      <div
        className="sc-bxivhb glCqXO"
      >
        <div
          className="Title sc-bdVaJa dLCEWS"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          simple
        </div>
        <div
          className="sc-kgoBCf hWwhUq"
        >
          &lt;My id="difool" /&gt;
          <div
            className="jsx"
          >
            &lt;My id="difool" /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH dNRbgB"
      >
        <div
          className="sc-gZMcBi hcMlfb"
        >
          I am John Difool
        </div>
      </div>
    </div>
  </div>
</div>
`;
